### DISCUSSION TITLE
open all applications in windows with cmd

### DISCUSSION THREAD

**Discussion by thanhngan22** [community/community#55323] [Programming Help]
### Select Topic Area

Question

### Body

Hi there,
I have a problem and need your help,

I want to create a program in cpp to do this thing:
- User enters the name of any application installed in their system, such as Word, Discord, notepad,  Camera, etc
- Then, my program will find the location of the executable file of this app and start that application.

This includes both default /system applications and apps that the user installed through cmd or the internet themselves.
For some apps I can start through this command after moving to the folder `Start Menu` : start + appName 
For system apps, I can start them after finding Name and InstallLocation when using Get-AppxPackage, for example, with Camera, I can start by running this command: start microsoft.windows.Camera:
For others  which users installed themselves, maybe I will iterate through folders and get the executable file and run it

But I don't know to determine which is an default app or an app installed by user to do later just through appName ???
In addition, also have a problem, an app with a name clock, when I find it's packageFamilyName, I got Microsoft.WindowsAlarms, another name Voice Recorder , I got Microsoft.WindowsSoundRecorder, what should I solve this ?

Do you have a better solution for this problem ?

Hope to get some help,
Thanks,

**Comment by anairamlc** [comment]
It sounds like you want to create a program that can start any application installed on a user's system. You can achieve this by using some of the built-in features of Windows PowerShell and C++.

To start an application using its name, you can use the ShellExecute function in C++. You can also use the PowerShell command "Get-AppxPackage" to find the location of system apps, and iterate through folders to find the executable file of user-installed apps.

Determining whether an app is a default app or a user-installed app can be tricky. One way to do this is to check whether the app is listed in the Windows registry under "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Model\Package Repository\Packages". If it is, then it is likely a default app.

As for the issue with the clock and voice recorder apps, it seems that their packageFamilyName is different from their display name. One way to solve this is to create a dictionary or lookup table that maps display names to packageFamilyNames.

Overall, this approach can be complex and time-consuming. If you're looking for a simpler solution, you may want to consider using a third-party library or tool that can handle this functionality for you.

**Reply by thanhngan22** [reply]
thanks for your help <3, I finished them by using PowerShell and Explorer to open applications then

**Comment by tuves** [comment]
Thanks for posting in the GitHub Community, @thanhngan22! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
